### DISCUSSION TITLE
creating Asyncstorage backup files

### DISCUSSION THREAD

**Discussion by panda0603** [react-native-async-storage/async-storage#511] [Q&A]
## You want to:  
Hello, 
I am creating a journal app. 

I want to enable users to backup their data manually by clicking 'backup' button, which will create a db backup file.
Upon reinstalling the app, users should be able to restore their data by importing the db backup file. 

Would this be possible? I've already seen posts about using Gdrive and ICloud, but this is automatic. I wish to export and import the db file of AsyncStorage manually. For example, expecting API something like AsyncStorage.createDbFile or equivalent. 
 
Alternative would be to JSON.stringify all the data in AsynStorage and put that on a txt file and export that, but I don't think this is a good idea ( or is it perfectly fine to do this?)

Thank you!

**Comment by bicanezin** [comment]
Any solutions?

**Comment by cs-manughian** [comment]
@panda0603 I'm wondering the same thing. But similarly to what you mentioned, a work-around would be to just do a `AsyncStorage.getItem` for what you want to back up and do it manually/programmatically. Then when a backup is pulled down, do a `AsyncStorage.setItem` for the proper keys. You could store it in whatever format you want in your file. To simplify the process, you could use `getAllKeys` or create a data structure of the keys, and just iterate over them and possibly their data types to get/set them.

Alternatively, it looks like AsyncStorage uses SQLite (based on OS) under the hood, so you could consider using that and then backing up the db file:
https://react-native-async-storage.github.io/async-storage/docs/advanced/where_data_stored

Here is a link to database backup exclusion also for reference (AsyncStorage disables that feature by default, which is likely desirable for a manual backup as well): 
https://react-native-async-storage.github.io/async-storage/docs/advanced/backup/
